python/pyarrow/tests/test_compute.py

```diff
@@ -1187,8 +1187,8 @@ def test_drop_null(ty, values):
 
 
 def test_drop_null_chunked_array():
-    arr = pa.chunked_array([["a", None], ["c", "d", None], [None], []])
-    expected_drop = pa.chunked_array([["a"], ["c", "d"], [], []])
+    arr = pa.chunked_array([[None, None], [None], []])
```

Comment:
Not sure why you're changing this test?

Reply:
In C++, a ChunkedArray has to have all Arrays of the same type, but [the Python API allowed coercion of `None`](https://github.com/apache/arrow/pull/12096/files#diff-cede36e8e2e0eb6e6e1ee21745db9687174527f463520c6e6d8b9e8f957bf304L1313)
```c++
["a", None] --> StringType
[None] --> NullType
[] --> NullType
```

Reply:
Hmm, this could be a slight API break. @jorisvandenbossche What do you think?

Reply:
Let me double check carefully the current C++ and Python behavior before jumping to conclusions.